datastore/concepts.js

```diff
@@ -1040,33 +1040,37 @@ Query.prototype.testEventualConsistentQuery = function () {
 
 // [START transactional_update]
 function transferFunds (fromKey, toKey, amount, callback) {
-  var error;
+  var transaction = datastore.transaction();
+
+  transaction.run(function (err) {
+    if (err) {
+      return callback(err);
+    }
 
-  datastore.runInTransaction(function (transaction, done) {
     transaction.get([
       fromKey,
       toKey
     ], function (err, accounts) {
       if (err) {
-        // An error occurred while getting the values.
-        error = err;
-        transaction.rollback(done);
-        return;
+        return transaction.rollback(function (_err) {
+          return callback(err || _err);
```

Comment:
No, I think we should prefer sending to the user the reason the transaction failed, instead of the reason the rollback failed. Or maybe send both? Concatenate the error messages.